### PR TITLE
Add GHMHD fd time derivative

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Only "Add GH+GRMHD FD time derivative" commit is new

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/TimeDerivative.hpp [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+          get<gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataVector>>(
+              temp_tags)
+              .get(i, j) =
+              0.5 * (pi.get(i + 1, j + 1) + phi_one_normal.get(i, j + 1) +
+                     phi_one_normal.get(j, i + 1));
+        }
+      }
+    }  // End scope for computing metric terms in GRMHD source terms.
+
+    // GeneralizedHarmonic::spatial_deriv_of_shift(
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/TimeDerivative.hpp]: remove

File: tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_TimeDerivative.cpp [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+      std::move(domain), std::move(external_boundary_conditions),
+      dummy_volume_mesh_velocity, dummy_normal_covector_and_magnitude, time,
+      clone_unique_ptrs(functions_of_time), soln, DummyEvolutionMetaVars{},
+      // Note: These damping functions all assume Grid==Inertial. We need to
+      // rescale the widths in the Grid frame for binaries.
+      std::unique_ptr<DampingFunction>(  // Gamma0, taken from SpEC BNS
+          std::make_unique<GeneralizedHarmonic::ConstraintDamping::
+                               GaussianPlusConstant<3, Frame::Grid>>(
+              0.01, 0.09 * 1.0 / 1.4, 5.5 * 1.4, std::array{0.0, 0.0, 0.0})),
+      std::unique_ptr<DampingFunction>(  // Gamma1, taken from SpEC BNS
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/Test_TimeDerivative.cpp]: move comments to where gamma1(2) are defined

**Review by kidder — COMMENTED** [review]
squash

**Comment by nilsdeppe** [comment]
Done and rebased. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**